relay-server/src/service.rs

```diff
@@ -356,8 +370,14 @@ impl ServiceState {
     }
 
     /// Returns the V2 envelope buffer, if present.
-    pub fn envelope_buffer(&self) -> Option<&ObservableEnvelopeBuffer> {
-        self.inner.registry.envelope_buffer.as_ref()
+    pub fn envelope_buffer(
```

Comment:
I would change the interface of `PartitionedEnvelopeBuffer` to behave roughly like `ObservableEnvelopeBuffer` behaved before. I.e. give it a `send(..)` and a `has_capacity()` method, but not expose internals about partitioning.

Reply:
I thought about this when designing the API, but it conflicted with how we access the buffer in the `queue_envelope`. I wanted to keep that logic like it is right now.